lib/lightning_web/live/workflow_live/components.ex

```diff
@@ -372,26 +373,41 @@ defmodule LightningWeb.WorkflowLive.Components do
       phx-change="validate"
       id="workflow_name_form"
     >
-      <div class="relative">
-        <%= text_input(
-          f,
-          :name,
-          class: "peer block w-full
-            text-2xl font-bold text-secondary-900
-            border-0 py-1.5 focus:ring-0",
-          required: true,
-          placeholder: "Untitled"
-        ) %>
-        <div
-          class="absolute inset-x-0 bottom-0
-                 peer-hover:border-t peer-hover:border-gray-300
-                 peer-focus:border-t-2 peer-focus:border-indigo-600
-                 peer-invalid:border-t-2 peer-invalid:border-rose-400"
-          aria-hidden="true"
-        >
-        </div>
+      <div class="relative flex items-center space-x-4">
+        <%= if @editing do %>
```

Comment:
We shouldn't be storing the editing state on the server. We can use the `focus` pseudo selector for that.

If you look at the diff, the previous approach was to use the `peer` facility of Tailwind to control the visibility of the border at the bottom.

```
<div
  class="absolute inset-x-0 bottom-0
         peer-hover:border-t peer-hover:border-gray-300
         peer-focus:border-t-2 peer-focus:border-indigo-600
         peer-invalid:border-t-2 peer-invalid:border-rose-400"
   aria-hidden="true"
>...
```

Where the text input had the `peer` class on it... so the elements are next to each other but hovering and focus states on the text input would control the line at the bottom.

I suggest using this approach with the pencil icon, i.e. render it all the time, but have it's visibility controlled by the `peer-focus:...` pseudo selectors.

And then using `phx-click` use the `JS` helpers on LiveView to set the focussed element to the text input.